ShadowDOM/explainer.md

```diff
@@ -0,0 +1,395 @@
+            height: 100vh;
+            background-color: #f0f0f0;
+        }
+        button, input[type="range"] {
+            cursor: pointer;
+            margin: 5px;
+        }
+    </style>
+</head>
+```
+Meanwhile, the styles defined within the Shadow DOM are specific to the media control widget. These styles ensure that the widget looks consistent and isn't affected by other styles on the page.
+```js
+class MediaControl extends HTMLElement {
+    constructor() {
+        super();
+        
+        // Attach a shadow root to the element.
+        const shadow = this.attachShadow({ mode: 'open' });
+
+        // Create elements
+
+        // Style the elements within the shadow DOM
+        const style = document.createElement('style');
+        style.textContent = `
+            .media-control-container {
+                display: flex;
+                flex-direction: column;
+                align-items: center;
+                border: 1px solid #ccc;
+                padding: 16px;
+                background-color: #fff;
+            }
+            .controls {
+                margin-top: 8px;
+                display: flex;
+                gap: 8px;
+                align-items: center;
+            }
+            button, input[type="range"] {
+                cursor: pointer;
+                margin: 5px;
+            }
+        ;
+    }
+}
+```
+Both the controls in the parent document and the controls inside the media control widget share the same base styles for cursor and margin. 
+
+### Anywhere web components are used
+When asked about pain points in [Web Components](https://2023.stateofhtml.com/en-US/features/web_components/), the number one issue, with 13% of the vote, is styling and customization. Many respondents specifically mentioned the difficulty of style sharing issues within the shadow DOM:
+* "I want to use shadow DOM to keep the light DOM tidy and use slots, but I don't always want style isolation"
+* "Inheriting/passing CSS styles from the main DOM to a shadow DOM"
+* "Shadow dom is a nightmare due to inability to style with global styles"
+* "I love to write my custom web components. It is supper easy to write, maintain. It organizes project structure in some small chunks. But I don't use shadow dom, because of css styles which i don't know how to share between web components"
+* "Shadow DOM encapsulation is too much. E.g. No way to adopt form styling from the surrounding page for common elements (buttons, inputs, etc) unless I'm willing to put them in light DOM"
+
+For additional use cases, please see issue [939](https://github.com/WICG/webcomponents/issues/939) 
+
+## Alternatives to using style in DSD 
+### Constructable Stylesheets 
+Developers can create stylesheets that can be applied to multiple shadow roots, using existing JavaScript, as outlined by the example below. 
+
+Step 1: Create a new Constructable Stylesheet: 
+```js
+const constStylesheet = new CSSStyleSheet(); 
```

Comment:
Suggest `constructableStylesheet` for the variable name so the meaning isn't confused with "constant stylesheet"